### PR TITLE
feat: crash tests

### PR DESCRIPTION
**Pull request by debendraoli**
### Commit Message

```bash
Crashes nodes to test relayer
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
## [0.1.0] - 2023-08-18
### Added

- Crash Relay On Any Chain Unreachable and Restore Relay
- Integrate Mock Ibc Dapp with integration tests

```

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the unit tests
- [x] I only have one commit (if not, squash them into one commit).
- [x] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: test/e2e/tests/relayer.go [hunk 1]
```diff
@@ -66,9 +66,29 @@ func (r *RelayerTestSuite) TestRelayer(ctx context.Context) {
...
+		r.PacketFlow(ctx, chainA, chainB, "test")
+	})
+
+	r.T.Run("crash nodes", func(t *testing.T) {
+		chainA, chainB := r.GetChains()
+		r.Require().NoError(r.CrashNode(ctx, chainA))
+		r.Require().NoError(r.CrashNode(ctx, chainB))
+	})
+
+	r.T.Run("test crash and recover relay", func(t *testing.T) {
```
Comment (AntonAndell) [inline test/e2e/tests/relayer.go:80]: To make this test more realisitic i would want packets to be sent while the relayer is down,
Prefarably, sent, acknowledged and timed out and would like to see it recover

**Review by AntonAndell — CHANGES_REQUESTED** [review]
Structure looks good but crash scenarios could be more thorough

File: test/e2e/tests/relayer.go (hunk 1 above)
Comment (debendraoli) [reply test/e2e/tests/relayer.go]: Improved.

Comment (AntonAndell) [reply test/e2e/tests/relayer.go:94]: Not sure i understand? 
isnt the flow now just
send packet and relay 
Close relay
start Relay
ExecuteCall

Which means the relay does not anything after recovery?

Comment (debendraoli) [reply test/e2e/tests/relayer.go:94]: - Crash Both nodes, chainA and B
- Send Packets to both nodes
- Crash Relay
- Recover both chain's node
- Recover Relay
- Get sent packets before crash
- Ping Pong cross chain test

Comment (AntonAndell) [reply test/e2e/tests/relayer.go:94]: Aah okey so relay never get notified sice nodes are down when you publish the inital package?

Comment (debendraoli) [reply test/e2e/tests/relayer.go:94]: I am supposed to crash destination chain instead of both.

Comment (debendraoli) [reply test/e2e/tests/relayer.go]: Anton I have pushed the changes that may satisfies you're talking about?

File: test/chains/icon/localnet.go [hunk 2]
```diff
@@ -509,14 +537,14 @@ func (c *IconLocalnet) FindCallMessage(ctx context.Context, startHeight int64, f
 	block, _ := c.getFullNode().Client.GetBlockByHeight(&icontypes.BlockHeightParam{Height: icontypes.NewHexInt(int64(intHeight - 1))})
 	i, _ := event.Index.Int()
 	tx := block.NormalTransactions[i]
-	trResult, _ := c.getFullNode().TransactionResult(ctx, string(tx.TxHash))
+	trResult, _ := c.getFullNode().TransactionResult(context.WithoutCancel(ctx), string(tx.TxHash))
```
Comment (redlarva) [inline test/chains/icon/localnet.go:540]: Is there a "WithoutCancel" method present within the context?

**Review by ibrizsabin — APPROVED** [review]

**Review by AntonAndell — APPROVED** [review]
Test logic looks good now

**Pull request merged**